gadget/gadgettest/examples.go

```diff
@@ -0,0 +1,931 @@
+	Structure: []gadget.OnDiskStructure{
+		{
+			LaidOutStructure: gadget.LaidOutStructure{
+				VolumeStructure: &gadget.VolumeStructure{
+					Name: "BIOS Boot",
+					Type: "21686148-6449-6E6F-744E-656564454649",
+					ID:   "b2e891ee-b971-4a2b-b874-694bbf9b821a",
+					Size: 1048576,
+				},
+				StartOffset: 1048576,
+			},
+			DiskIndex: 1,
+			Node:      "/dev/sda1",
+			Size:      1048576,
+		},
+		{
+			LaidOutStructure: gadget.LaidOutStructure{
+				VolumeStructure: &gadget.VolumeStructure{
+					Name:       "EFI System",
+					Type:       "C12A7328-F81F-11D2-BA4B-00A0C93EC93B",
+					ID:         "a87e9dcb-b1e1-4eab-89cf-1c2fc057b038",
+					Label:      "system-boot",
+					Filesystem: "vfat",
+					Size:       52428800,
+				},
+				StartOffset: 2097152,
+			},
+			DiskIndex: 2,
+			Node:      "/dev/sda2",
+			Size:      52428800,
+		},
+		{
+			LaidOutStructure: gadget.LaidOutStructure{
+				VolumeStructure: &gadget.VolumeStructure{
+					Name:       "writable",
+					Type:       "0FC63DAF-8483-4772-8E79-3D69D8477DE4",
+					ID:         "cba2b8b3-c2e4-4e51-9a57-d35041b7bf9a",
+					Label:      "writable",
+					Filesystem: "ext4",
+					Size:       10682875392,
+				},
+				StartOffset: 54525952,
+			},
+			DiskIndex: 3,
+			Node:      "/dev/sda3",
+			Size:      10682875392,
+		},
+	},
+	ID:               "2a9b0671-4597-433b-b3ad-be99950e8c5e",
+	Device:           "/dev/sda",
+	Schema:           "gpt",
+	Size:             10737418240,
+	UsableSectorsEnd: 20971487,
+	SectorSize:       512,
+}
+
+var UC16ImplicitSystemDataMockDiskMapping = &disks.MockDiskMapping{
+	DevNode: "/dev/sda",
+	DevPath: "/sys/devices/pci0000:00/0000:00:01.1/ata1/host0/target0:0:0/0:0:0:0/block/sda",
+	DevNum:  "600:1",
+	// assume 34 sectors at end for GPT headers backup
+	DiskUsableSectorEnd: 10240*1024*1024/512 - 33,
+	DiskSizeInBytes:     10240 * 1024 * 1024,
+	SectorSizeBytes:     512,
+	DiskSchema:          "gpt",
+	ID:                  "f69dbcfe-1258-4b36-9d1f-817fdeb61aa3",
+	Structure: []disks.Partition{
+		{
+			KernelDeviceNode: "/dev/sda1",
+			KernelDevicePath: "/sys/devices/pci0000:00/0000:00:01.1/ata1/host0/target0:0:0/0:0:0:0/block/sda/sda1",
+			PartitionUUID:    "420e5a20-b888-42e2-b7df-ced5cbf14517",
+			PartitionLabel:   "BIOS\\x20Boot",
+			PartitionType:    "21686148-6449-6E6F-744E-656564454649",
+			StartInBytes:     1024 * 1024,
+			SizeInBytes:      1024 * 1024,
+			Major:            600,
+			Minor:            2,
+			DiskIndex:        1,
+		},
+		{
+			KernelDeviceNode: "/dev/sda2",
+			KernelDevicePath: "/sys/devices/pci0000:00/0000:00:01.1/ata1/host0/target0:0:0/0:0:0:0/block/sda/sda2",
+			PartitionUUID:    "fc8626b9-af30-4b3c-83c4-05bed39bb82e",
+			PartitionLabel:   "EFI\\x20System",
+			PartitionType:    "C12A7328-F81F-11D2-BA4B-00A0C93EC93B",
+			FilesystemUUID:   "6D21-B3FE",
+			FilesystemLabel:  "system-boot",
+			FilesystemType:   "vfat",
+			StartInBytes:     (1 + 1) * 1024 * 1024,
+			SizeInBytes:      50 * 1024 * 1024,
```

Comment:
oneMeg

That `(1 + 1)` looks a bit funny, but it's OK, one never knows who the reader will be ;-)

Reply:
I found it useful to see why it's 2, since it's offset + size of previous structures, though for the second structure since it's always just 1+1 it is admittedly less useful

Reply:
Just teasing :-)